Add and list student prizes

dawidkaplon opened the pull request:
### Two extra endpoints that manages specific student's prizes have been added.  We can now:

1. List all prizes of a student using `GET api/students/{id}/prizes/`
2. Add new prize using `POST api/students/{id}/prizes/`

Please review and run this code on your machines and let me know if you have any objections.

Commit:
Add an endpoint to list prizes of individual students

Implement a POST endpoint for adding prizes to individual students

MateuszBelczowski commented on users/resources.py:
```diff
@@ -104,6 +104,33 @@ def delete(self, request, pk):
+    def get(self, request, pk):
+        user_id = request.user.id
+
+        if not has_user_access_to_student(user_id, pk):
```
It's not always necessary to declare variable if we're using it only at one place. As an alternative we could write it as

```suggestion
        if not has_user_access_to_student(request.user.id, pk):
```

but I'll leave the decision up to you

MateuszBelczowski commented on users/resources.py:
```diff
@@ -104,6 +104,33 @@ def delete(self, request, pk):
+        if not has_user_access_to_student(user_id, pk):
+            raise PermissionDenied
+
+        student = Student.objects.get(pk=pk)
+        prizes = Prize.objects.filter(student=student)
```
It's not necessary to retrieve student object to later use it in the filtering. Filtering can be done directly using the id value
```suggestion
        prizes = Prize.objects.filter(student=pk)
```

or 

```suggestion
        prizes = Prize.objects.filter(student_id=pk)
```

I believe both options should work

MateuszBelczowski commented on users/serializers.py:
```diff
@@ -40,7 +40,9 @@ class PrizeSerializer(serializers.Serializer):
     def create(self, validated_data):
-        return Prize.objects.create(**validated_data)
+        student_data = validated_data.pop('student')
+        student = Student.objects.get(pk=student_data['id'])
```
I think the same as above applies here. If I recall correctly you can use id and don't need to retrieve the whole object.

MateuszBelczowski commented on users/resources.py:
```diff
@@ -104,6 +104,33 @@ def delete(self, request, pk):
         return Response(status=status.HTTP_204_NO_CONTENT)
 
 
+class StudentPrizesResource(APIView):
```
I think that we should keep names consistent - other views are named e.g. `PointResource` and not `StudentPointsResource` (so beside the `student` prefix, please also note single vs plural). For now I'd stick to this convention, we can always adjust later

MateuszBelczowski reviewed (commented):
Overall, it looks good, but left a few minor comments. There are ways to simplify this code, e.g.

- write custom permission class to check if user has access to student
- we don't always need to override full methods (like `get`, `post`)

but we're not doing so in other places, so I'd stick to this approach for now and once we get it working, we can refactor it a little bit. Please also note that we're not using pagination here - not sure if that's expected or not

Commit:
Create HasUserAccessToStudent custom permission class

Replace the old PrizesResource class with a new one, focused on connecting prizes and students

Commit:
Merge branch 'develop' into add-and-list-student-prizes

Commit:
Reformat the code to pass pre-commit tests
